api/src/database/migrations/20210903A-add-auth-provider.ts

```diff
@@ -0,0 +1,21 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_users', (table) => {
+		table.string('provider', 128).notNullable().defaultTo('default');
+		table.string('alternate_identifier', 512).unique();
```

Comment:
I'd have said email is the identifier at present, at least for auth. I chose `alternate_identifier` because we wanted support usernames for local, but since that's changed we can use `external_identifier`.